app/src/main/java/io/homeassistant/companion/android/sensors/BatterySensorManager.kt

```diff
@@ -28,60 +29,117 @@ class BatterySensorManager : SensorManager {
+                retVal.add(
+                    SensorRegistration(
+                        sensor,
+                        "Battery State",
+                        "battery"
+                    )
+                )
+            }
+
             return@let retVal
         } ?: listOf()
     }
 
     override fun getSensors(context: Context): List<Sensor<Any>> {
         val retVal = ArrayList<Sensor<Any>>()
+
         context.registerReceiver(null, IntentFilter(Intent.ACTION_BATTERY_CHANGED))?.let {
             getBatteryLevelSensor(it)?.let { sensor ->
                 retVal.add(sensor)
             }
+
+            getBatteryStateSensor(it)?.let { sensor ->
+                retVal.add(sensor)
+            }
         }
 
         return retVal
     }
 
-    private fun getBatteryLevelSensor(intent: Intent): Sensor<Any>? {
-        val level = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1)
-        val scale = intent.getIntExtra(BatteryManager.EXTRA_SCALE, -1)
-        val status: Int = intent.getIntExtra(BatteryManager.EXTRA_STATUS, -1)
-
-        if (level == -1 || scale == -1) {
-            Log.e(TAG, "Issue getting battery level!")
-            return null
-        }
+    private fun getBatteryIcon(batteryStep: Int, isCharging: Boolean = false, chargerType: String? = null, chargingStatus: String? = null): String {
+        var batteryIcon = "mdi:battery"
 
-        val isCharging: Boolean = status == BatteryManager.BATTERY_STATUS_CHARGING ||
-                status == BatteryManager.BATTERY_STATUS_FULL
+        if (chargingStatus == "unknown") {
+            batteryIcon += "-unknown"
 
-        val chargerType = when (intent.getIntExtra(BatteryManager.EXTRA_PLUGGED, -1)) {
-            BatteryManager.BATTERY_PLUGGED_AC -> "AC"
-            BatteryManager.BATTERY_PLUGGED_USB -> "USB"
-            BatteryManager.BATTERY_PLUGGED_WIRELESS -> "Wireless"
-            else -> "N/A"
+            return batteryIcon
         }
 
-        val percent = (level.toFloat() / scale.toFloat() * 100.0f).toInt()
-        var batteryIcon = "mdi:battery"
         if (isCharging)
             batteryIcon += "-charging"
-        if (chargerType == "Wireless")
+
+        if (chargerType == "wireless")
             batteryIcon += "-wireless"
 
-        val batteryStep = percent / 10
         batteryIcon += when (batteryStep) {
             0 -> "-outline"
             10 -> ""
             else -> "-${batteryStep}0"
         }
 
+        return batteryIcon
+    }
+
+    private fun getBatteryLevelSensor(intent: Intent): Sensor<Any>? {
+        val level: Int = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1)
+        val scale: Int = intent.getIntExtra(BatteryManager.EXTRA_SCALE, -1)
+
+        if (level == -1 || scale == -1) {
+            Log.e(TAG, "Issue getting battery level!")
+            return null
+        }
+
+        val percentage: Int = (level.toFloat() / scale.toFloat() * 100.0f).toInt()
```

Comment:
Extract out the building of the battery percentage so we don't have to duplicate it.